src/Microsoft.CorrelationVector/CorrelationVectorV2.cs

```diff
@@ -0,0 +1,338 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Globalization;
+
+namespace Microsoft.CorrelationVector
+{
+    /// <summary>
+    /// This class represents a lightweight vector for identifying and measuring
+    /// causality.
+    /// </summary>
+    public sealed class CorrelationVectorV2 : CorrelationVector
+    {
+        internal new const byte MaxVectorLength = 127;
+        internal const byte BaseLength = 22;
+
+        internal readonly string BaseVector = null;
+
+        private int extension = 0;
+
+        private bool immutable = false;
+
+        private static Random rng = new Random();
+
+        /// <summary>
+        /// This is the termination sign which should be used when vector length exceeds 
+        /// the <see cref="MaxVectorLength"/>
+        /// </summary>
+        public const string TerminationSign = "!";
+
+        /// <summary>
+        /// Creates a new correlation vector by extending an existing value. This should be
+        /// done at the entry point of an operation.
+        /// </summary>
+        /// <param name="correlationVector">
+        /// Taken from the message header indicated by <see cref="HeaderName"/>.
+        /// </param>
+        /// <returns>A new correlation vector extended from the current vector.</returns>
+        public new static CorrelationVectorV2 Extend(string correlationVector)
+        {
+            if (CorrelationVectorV2.IsImmutable(correlationVector))
+            {
+                return CorrelationVectorV2.Parse(correlationVector);
+            }
+
+            if (CorrelationVectorV2.ValidateCorrelationVectorDuringCreation)
+            {
+                CorrelationVectorV2.Validate(correlationVector);
+            }
+
+            if (CorrelationVectorV2.IsOversized(correlationVector, 0))
+            {
+                return CorrelationVectorV2.Parse(correlationVector + CorrelationVectorV2.TerminationSign);
+            }
+
+            return new CorrelationVectorV2(correlationVector, 0, false);
+        }
+
+        /// <summary>
+        /// Creates a new correlation vector by applying the Spin operator to an existing value.
+        /// This should be done at the entry point of an operation.
+        /// </summary>
+        /// <param name="correlationVector">
+        /// Taken from the message header indicated by <see cref="HeaderName"/>.
+        /// </param>
+        /// <returns>A new correlation vector extended from the current vector.</returns>
+        public new static CorrelationVectorV2 Spin(string correlationVector)
```

Comment:
Here too, we don't need this method

Reply:
I'd prefer people to be able to choose between calling CorrelationVector.Spin and CorrelationVectorV2.Spin if they so choose to

Reply:
We shouldn't expose that functionality since its not required. Not having CorrelationVectorV2.Spin and calling that method will result in CorrelationVector.Spin getting called and performing the right operations